Makefile

```
@@ -13,111 +15,104 @@ L=[01;30m
+	@ echo ""
+
+
+build_pilasweb: # For develop with Pilas Web WITHOUT releasing it.
 	cd ../pilasweb; make build
 	cp -rf ../pilasweb/dist node_modules/pilasweb/
 
-version_patch:
-	$(EMBER) release
-
-version_minor:
-	$(EMBER) release --minor
-
-version_major:
-	$(EMBER) release --major
+build_pilas_bloques_exercises: #For develop with Pilas Bloques Exercises WITHOUT releasing it.
+	@ echo "${G}Building Pilas Bloques Exercises${N}"
+	cd ../pilas-bloques-exercises; node_modules/grunt-cli/bin/grunt
+	cp -rf ../pilas-bloques-exercises/dist node_modules/pilas-bloques-exercises/
 
-empaquetar: build _preparar_electron _empaquetar_osx _empaquetar_win32 _empaquetar_linux
-	@echo "${G}Listo, los binarios se generaron en el directorio 'binarios'.${N}"
+watch_pilas_bloques_exercises: 
+	@ echo "${G}Building Pilas Bloques Exercises${N}"
+	cd ../pilas-bloques-exercises; node_modules/grunt-cli/bin/grunt watch
 
-_preparar_electron:
-	@echo "${G}Preparando directorio dist para funcionar con electron...${N}"
-	@cp package.json dist/package.json
-	@cp packaging/electron.js dist
+build: npm run build
 
-empaquetar = @echo "${G}Empaquetando binarios para $(1) $(2)...${N}"; node_modules/.bin/electron-packager dist ${NOMBRE} --app-version=${VERSION} --platform=$(1) --arch=$(2) --ignore=node_modules --out=binarios --overwrite --icon=packaging/icono.$(3)
-
-_empaquetar_osx:
-	rm -f binarios/${NOMBRE}-${VERSION}.dmg
-	$(call empaquetar,darwin,all,icns)
-	hdiutil create binarios/${NOMBRE}-${VERSION}.dmg -srcfolder ./binarios/${NOMBRE}-darwin-x64/${NOMBRE}.app -size 1g
+build_live:
+	$(EMBER) build --watch
 
-_empaquetar_win32:
-	$(call empaquetar,win32,ia32,ico)
-	@echo "${G}Generando instalador para windows...${N}"
-	cp packaging/instalador.nsi binarios/${NOMBRE}-win32-ia32/
-	cd binarios/${NOMBRE}-win32-ia32/; makensis instalador.nsi
-	@mv binarios/${NOMBRE}-win32-ia32/${NOMBRE}.exe binarios/${NOMBRE}-${VERSION}.exe
+release_patch:
+	$(EMBER) release
 
-_empaquetar_linux: _borrar_binarios_linux _empaquetar_zip_linux_x64 _empaquetar_zip_linux_ia32 _empaquetar_deb_linux_x64
+release_minor:
+	$(EMBER) release --minor
 
-_borrar_binarios_linux:
-	rm -rf binarios/${NOMBRE}-*linux*
-	rm -rf binarios/*.deb
+release_major:
+	$(EMBER) release --major
 
-# Este empaquetado tiene el problema de que NO reemplaza la aplicación vieja de Huayra.
-# Además, el package debian generado tiene nombre diferente al viejo.
-_empaquetar_deb_linux_x64:
-	$(call empaquetar,linux,x64,icns)
+package = @echo "${G}Generating package for $(1) $(2)...${N}"; node_modules/.bin/electron-packager dist ${NAME} --app-version=${VERSION} --platform=$(1) --arch=$(2) --ignore=node_modules --out=binarios --overwrite --icon=packaging/icono.$(3)
+
+electron_pre_build:
+	@ echo "${G}Making dist directory works with electron...${N}"
+	if ! ([ -d "./binaries" ]); then mkdir ./binaries; fi
+	cp package.json dist/package.json
+	cp packaging/electron.js dist
+
+electron_osx_package:
+	make electron_pre_build;
```

Make already handles target dependences. That's why it was used before like this: 

```suggestion
electron_osx_package: electron_pre_build
```